pkg/certificate/controller.go

```diff
@@ -33,9 +38,38 @@ func (m *Manager) add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return errors.Wrap(err, "failed instanciating certificate controller")
 	}
 
-	isWebhookConfigOrAnnotatedResource := func(meta metav1.Object) bool {
+	isACascadeDelete := func(deleteEvent event.DeleteEvent) bool {
+		namespace := deleteEvent.Meta.GetNamespace()
+		for _, ownerRefernce := range deleteEvent.Meta.GetOwnerReferences() {
+			gv, err := schema.ParseGroupVersion(ownerRefernce.APIVersion)
+			if err != nil {
+				logger.Error(err, "failed parsing gv from owner")
+				continue
+			}
+			gvk := gv.WithKind(ownerRefernce.Kind)
+			object := unstructured.Unstructured{}
+			object.SetGroupVersionKind(gvk)
+			err = m.client.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: ownerRefernce.Name}, &object)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					logger.Info("Cascade delete detected, at least one owner is not present")
+					return true
+				}
+				logger.Error(err, "failed detecting if this is a cascade delete")
+				continue
+			}
+		}
+		logger.Info("Cascade delete not-detected, all the owners are present")
+		return false
+	}
+
+	isAnnotatedResource := func(meta metav1.Object) bool {
 		_, foundAnnotation := meta.GetAnnotations()[secretManagedAnnotatoinKey]
-		return meta.GetName() == m.webhookName || foundAnnotation
+		return foundAnnotation
+	}
+
+	isWebhookConfigOrAnnotatedResource := func(meta metav1.Object) bool {
```

Comment:
This function alone does not bring any value. Better to have isAnnotatedResource and isWebhookConfig and then call them both with || in between.